### PR TITLE
v5.17.12

### PR DESCRIPTION
**Pull request by LukasTy**
### Prepare the release of the packages

The following steps must be proposed as a pull request.

- [x] Compare the last tag with the branch upon which you want to release (`next` for the alpha / beta releases and `master` for the current stable version).
      To do so, use `yarn release:changelog` The options are the following:

  ```bash
  yarn release:changelog
     --githubToken   YOUR_GITHUB_TOKEN (needs "public_repo" permission)
     --lastRelease   The release to compare against (default: the last one)
     --release       The branch to release (default: master)
  ```

  You can also provide the github token by setting `process.env.GITHUB_TOKEN` variable.

  In case of a problem, another method to generate the changelog is available at the end of this page.

- [x] Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
- [x] Update the root `package.json`'s version
- [x] Update the versions of the other `package.json` files and of the dependencies with `yarn release:version`.
- [x] Fix manually the package version in `x-date-pickers/package.json` and `x-date-pickers-pro/package.json`.
- [x] Open PR with changes and wait for review and green CI.
- [ ] Merge PR once CI is green, and it has been approved.

### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```sh
git push upstream master:docs-v5 -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v5)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v5` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,24 @@
...
+- 🌍 Improve Finnish (fi-FI) locale (#6859) @RainoPikkarainen
+
+### `@mui/x-data-grid@v5.17.12` / `@mui/x-data-grid-pro@v5.17.12` / `@mui/x-data-grid-premium@v5.17.12`
+
+#### Changes
+
+- [DataGridPremium] Support aggregating data from multiple row fields (#6844) @cherniavskii
+- [DataGridPro] Fix detail panel not working with `getRowSpacing` prop (#6858) @cherniavskii
+
+- [I10n] Improve Finnish (fi-FI) locale on the data grid (#6859) @RainoPikkarainen
```
Comment (flaviendelangle) [inline CHANGELOG.md:22]:
```suggestion
- [l10n] Improve Finnish (fi-FI) locale on the data grid (#6859) @RainoPikkarainen
```

**Review by flaviendelangle — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,24 @@
...
+- 🐞 Bugfixes
+- 🌍 Improve Finnish (fi-FI) locale (#6859) @RainoPikkarainen
+
+### `@mui/x-data-grid@v5.17.12` / `@mui/x-data-grid-pro@v5.17.12` / `@mui/x-data-grid-premium@v5.17.12`
+
+#### Changes
+
+- [DataGridPremium] Support aggregating data from multiple row fields (#6844) @cherniavskii
+- [DataGridPro] Fix detail panel not working with `getRowSpacing` prop (#6858) @cherniavskii
+
```
[the remark this answers is not in the document]
Comment (LukasTy) [reply CHANGELOG.md:21]: Will bring them in. Waiting for docs/monorepo PR to be merged. 🙈

File: .circleci/config.yml [hunk 3]
```diff
@@ -162,7 +162,7 @@ jobs:
       - install_js
       - run:
           name: '`yarn prettier` changes committed?'
-          command: yarn prettier check-changed
+          command: yarn prettier --check
```
Comment (LukasTy) [inline .circleci/config.yml:165]: @m4theushw 
I've also brought [this change](https://github.com/mui/mui-x/pull/6876/commits/c66564f7efc6d8ddf8636e324360e8af34c8033a) in as to not introduce a pipeline regression after the monorepo/prettier change. 😉

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (LukasTy) [reply CHANGELOG.md:22]: I've left the original PR title in the changelog. Do you think using your copyright is more correct?

[the remark this answers is not in the document]
Comment (LukasTy) [reply CHANGELOG.md:22]: Yes, I've added the original PR author. I'm only asking if I should do the following diff:
```diff
- [DataGrid] Fix conflict with `onResize` added to `React.HTMLAttributes` (#6883) @vizv
+ [DataGrid] Fix conflict with the latest version of `@types/react` (#6883) @vizv
```

File: packages/x-date-pickers-pro/package.json [hunk 4]
```diff
@@ -48,7 +48,7 @@
     "@date-io/moment": "^2.15.0",
     "@mui/utils": "^5.10.3",
     "@mui/x-date-pickers": "5.0.8",
-    "@mui/x-license-pro": "5.17.11",
+    "@mui/x-license-pro": "5.17.12",
```
Comment (LukasTy) [inline packages/x-date-pickers-pro/package.json:51]: After thinking about it a bit more—I think we need to release `@mui/x-date-pickers-pro` package even though it does not have any changes of it's own, but I reckon we don't want our users end up needing both `5.17.11` and `5.17.2` versions of `x-license-pro` installed if they are using both pro/premium pickers and data grid packages.

But then should we also release empty `@mui/x-date-pickers`?
Have we ever had such a release case?
What do you think @flaviendelangle @m4theushw @cherniavskii ?

Comment (LukasTy) [reply packages/x-date-pickers-pro/package.json:51]: Yeah, agreed. Will bump both the pickers packages for consistency.

Comment (cherniavskii) [reply packages/x-date-pickers-pro/package.json:51]: Interesting!

Currently, our users can have more than one version of the `x-license-pro`  package installed if they're using different versions of `x-data-grid` and `x-date-pickers`. I believe releasing `x-date-pickers` just to bump the license package version won't solve it.

Are there any known issues with multiple versions of `x-license-pro` installed though?

Comment (LukasTy) [reply packages/x-date-pickers-pro/package.json:51]: Not that I'm aware of, but if we can avoid having two versions by force if people are updating packages frequently—I'd be in favour of exercising such practice. 🤔

Comment (flaviendelangle) [reply packages/x-date-pickers-pro/package.json:51]: >  The license key will be set only for the root package.

The license key is stored in a global state so it should be shared across the instancves

**Review by cherniavskii — APPROVED** [review]

**Review by joserodolfofreitas — APPROVED** [review]

**Pull request merged**